src/xcsync/ExtensionFilter.cs

```diff
@@ -0,0 +1,26 @@
+namespace xcsync;
+#pragma warning disable IO0006 // Replace Path class with IFileSystem.Path for improved testability
+public interface IFilesystemEventFilter {
+	bool ProcessEvent (string path); // You will have to list each event
+	bool ProcessRenameEvent (string origin, string destination);
+}
+
+public class ExtensionFilter (IEnumerable<string> extensionsToMonitor) : IFilesystemEventFilter {
+	readonly HashSet<string> _extensionsToMonitor = new (extensionsToMonitor, StringComparer.OrdinalIgnoreCase);
+
+	public bool ProcessEvent (string path)
+	{
+		var extension = Path.GetExtension (path);
+		return _extensionsToMonitor.Contains (extension);
+	}
+
+	public bool ProcessRenameEvent (string origin, string destination)
+	{
+		var originExtension = Path.GetExtension (origin);
+		var destinationExtension = Path.GetExtension (destination);
+		return _extensionsToMonitor.Contains (originExtension) || _extensionsToMonitor.Contains (destinationExtension);
+	}
+
+	public string GetExtensionsToMonitorAsString () => string.Join (", ", _extensionsToMonitor.Select (ext => ext));
```

Comment:
Works, but a little complicated, right?
```suggestion
	public string GetExtensionsToMonitorAsString () => string.Join (", ", _extensionsToMonitor);
```
string.Join takes an IEnumerable<T> ([docs](https://learn.microsoft.com/en-us/dotnet/api/system.string.join?view=net-8.0#system-string-join(system-string-system-collections-generic-ienumerable((system-string))))) and [HashSet](https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.hashset-1?view=net-8.0) implements IEnumerable.